mocks/mocks.go

```diff
@@ -364,7 +364,7 @@ func (sa *StorageAuthority) PreviousCertificateExists(
 }
 
 func (sa *StorageAuthority) GetPendingAuthorization(ctx context.Context, req *sapb.GetPendingAuthorizationRequest) (*core.Authorization, error) {
-	return nil, fmt.Errorf("GetPendingAuthorization not implemented")
+	return nil, berrors.NotFoundError("no pending authorization found")
```

Comment:
This seems like an odd change; was it needed to fix tests? Would it make sense to implement a more specific subclass of the SA mock that always returns a "not found" error?

Reply:
Yup, this broke any test that used `NewAuthorization` in the RA because it calls `GetPendingAuthorization` (which was previously inside a `features.ReusePendingAuthz` check). Not super convinced either way on having a new mock, it didn't require any extra fiddling to get other tests to work beyond this so ¯\_(ツ)_/¯.

Reply:
Oh, also the reason I chose a not found error was because that was what made the most sense in terms of `NewAuthorization` using it.

Reply:
The reason I've tended to prefered specialized subclasses lately is that they make it a lot more explicit which behaviors the unittest is depending on. Otherwise it gets all spooky-action-at-a-distance-y.